.devcontainer/devcontainer.json

```diff
@@ -13,23 +13,42 @@
 	"runArgs": [ "--cap-add=SYS_PTRACE", "--security-opt", "seccomp=unconfined" ],
 
 	// Set *default* container specific settings.json values on container create.
-	"settings": {
-		// GO
-		"go.toolsManagement.checkForUpdates": "local",
-		"go.useLanguageServer": true,
-		"go.gopath": "/go",
-		"go.goroot": "/usr/local/go",
-
-		// Misc
-		"trailing-spaces.trimOnSave": true
-	},
+	"customizations": {
+		"vscode": {
+			"settings": {
+				// GO
+				"go.toolsManagement.checkForUpdates": "local",
+				"go.useLanguageServer": true,
+				"go.gopath": "/go",
+				"go.goroot": "/usr/local/go",
 
-	// Add the IDs of extensions you want installed when the container is created.
-	"extensions": [
-		"golang.Go",
-		"graphql.vscode-graphql",
-		"rust-lang.rust-analyzer"
-	],
+				// Formatting
+				"editor.formatOnSave": true,
+				"files.trimTrailingWhitespace": true,
+				"files.insertFinalNewline": true,
+				"files.trimFinalNewlines": true,
+
+				// Misc
+				"vsicons.dontShowNewVersionMessage": true,
+				"workbench.iconTheme": "vscode-icons",
+				"git.autofetch": true
+			},
+
+			// Add the IDs of extensions you want installed when the container is created.
+			"extensions": [
+				"DavidAnson.vscode-markdownlint",
+				"golang.Go",
+				"graphql.vscode-graphql",
+				"GraphQL.vscode-graphql-syntax",
+				"HashiCorp.terraform",
+				"rust-lang.rust-analyzer",
+				"sourcegraph.sourcegraph",
+				"vscode-icons-team.vscode-icons",
+				"xaver.clang-format",
```

Comment:
What are we using the clang format extension for?

Reply:
Protobuf